dle/data/management/commands/remove_non_nda_dls_fda.py

```diff
@@ -0,0 +1,130 @@
+class Command(BaseCommand):
+    help = """Removes non-NDA drug labels from OpenFDA data. Should be a one-time run to clean up data,
+    as future runs will already filter to only include NDA (innovator) labels.
+    """
+
+    def __init__(self, stdout=None, stderr=None, no_color=False, force_color=False):
+        self.root_dir = settings.MEDIA_ROOT / "fda"
+        os.makedirs(self.root_dir, exist_ok=True)
+        super().__init__(stdout, stderr, no_color, force_color)
+
+    def add_arguments(self, parser):
+        parser.add_argument("--cleanup", type=strtobool, help="Set to cleanup files", default=False)
+
+    def handle(self, *args, **options):
+        self.cleanup = options["cleanup"]
+
+        # Download and extract JSON data if it doesn't exist
+        dl_json = json.loads(requests.get(FDA_JSON_URL).text)
+        labels_json = dl_json["results"]["drug"]["label"]
+        urls = [x["file"] for x in labels_json["partitions"]]
+        json_zips = self.download_json(urls)
+        self.extract_json_zips(json_zips)
+        file_dir = self.root_dir / "record_zips"
+
+        # Iterate the json files in the directory one by one
+        # Build a list of records to delete
+        for file in os.listdir(file_dir):
+            json_file = file_dir / file
+            raw_json_result = None
+            with open(json_file, encoding="utf-8-sig") as f:
+                raw_json_result = json.load(f)
+                logger.info(f"start loading json {json_file}")
+            raw_json_result = raw_json_result["results"]
+            logger.info(f"Finished loading {json_file}")
+
+            # Filter out non-NDA labels
+            logger.info("Filtering non-NDA labels")
+            records_to_delete = self.filter_data(raw_json_result)
+            logger.info(f"Records to delete: {len(records_to_delete)}")
+            logger.info("Sample --------")
+            logger.info(records_to_delete[0:100])
+            # Delete the records
+            break
+            # self.delete_data(records_to_delete)
+            # logger.info(f"Finished deleting {json_file}")
+
+    def filter_data(self, raw_json_result):
+        # create a list of records to delete
+        # only records that start with NDA should be kept
+        app_nums_to_delete = []
+        for record in raw_json_result:
+            try:
+                application_num = record["openfda"]["application_number"]
+                if not application_num.startswith("NDA"):
+                    app_nums_to_delete.append(application_num)
+            except KeyError:
+                logger.info(f"KeyError - no application_number: {record['openfda']}")
+        return app_nums_to_delete
+
+    def delete_data(self, records):
+        # delete the records
+        for record in records:
+            application_num = record["metadata"]["application_number"]
+            DrugLabel.objects.filter(source_product_number=application_num).delete()
```

Comment:
I don't think we are using the application number as source product number?

Reply:
Yep totally right. It's the `product_ndc` which we were messaging about in Slack, but we need to check the application number for whether to delete or not which got me confused